### PR TITLE
feat: generate seo sitemap on build (#819)

### LINKED ISSUE #819: Create test for HCAGetDataFileSummaryComponent component

### LINKED ISSUE #819: [AnVIL DX] Create an SEO site map (Google Site map xml file) as part of the build
#### Goal
All pages have a sitemap entry similar to:
https://data.humancellatlas.org/explore/projects-sitemap.xml

- Detail Pages like https://hca-data-portal.dev.clevercanary.com/explore/projects/60ea42e1-af49-42f5-8164-d641fdb696bc
- Detail Page Tabs like https://hca-data-portal.dev.clevercanary.com/explore/projects/60ea42e1-af49-42f5-8164-d641fdb696bc/project-matrices
- Entity lists like https://hca-data-portal.dev.clevercanary.com/explore/projects

```[tasklist]
### Tasks
- [x] Review approaches, and present one or two for a decision.
- [ ] Implement
```

### PR DESCRIPTION
**Pull request by BruceRodrigues**
### Ticket
Closes clevercanary/data-browser#819

### Reviewers
@

### Changes
- Added the generation of sitemap.xml during build
- Removed the need to create scripts for each site config, that way each site config could be run using an npm argument. E.g. `npm run dev --site=hca-dcp`

### Known Issues
next-sitemap doesn't have support for typescript so we can't just import a typescript file inside `next-sitemap.config.mjs`. To do that, we would need to transpile into js before importing, the problem of doing that is that if we try to only transpile `app/config` we'll ending having to transpile a lot of files, since it imports from every config, including images, which will make the import even harder to do, since `tsc` won't add the images on the bundle.

I can think of a way to simplify this and also get rid of the need for the `CONFIGS` map, which is using dynamic imports. But I'll wait for more directions before going too deep into this solution, since having `sitemap` is currently a blocker.

### CODE REVIEW

File: explorer/package.json [hunk 1]
```diff
@@ -3,20 +3,9 @@
   "version": "0.1.0",
   "private": true,
   "scripts": {
-    "dev:ncpi-catalog-dug": "./scripts/dev.sh ncpi-catalog-dug && next dev",
```
Comment (NoopDog) [inline explorer/package.json:6]: Hi @BruceRodrigues, this is interesting but let's create a separate ticket for this. I have build scripts that rely on these at the moment and I would like to not have to update them for now. Can you revert this change that removes all the scripts? Thanks!

File: explorer/next-sitemap.config.mjs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/* eslint-disable sonarjs/no-duplicate-string -- comment */
+const URLS = {
+  "anvil-catalog-dev": "https://anvil-portal.dev.clevercanary.com",
```
Comment (NoopDog) [inline explorer/next-sitemap.config.mjs]: Instead of this, we can just take it from the environment directly and modify .env for each environment. 

LIke siteUrl: process.env.NEXT_PUBLIC_SITEMAP_DOMAIN]

Thanks!

**Review by NoopDog — CHANGES_REQUESTED** [review]
Thanks for this Bruce. Let's keep the scripts how they are for the time being as I have other scripts that use them that I don't want to modify at the moment. 

Also, I can drop the requirement that we need to take the config to form our site configs. To simplify this let's just put the sitemap root URL in the .env files per environment.

Cheers,
D

File: explorer/package.json (hunk 1 above)
Comment (BruceRodrigues) [reply explorer/package.json:6]: Sure thing!

**Comment by BruceRodrigues** [comment]
I just noticed that this PR is pointing to DataBiosphere, will close this one.
This is the one pointing to cleverCanary https://github.com/clevercanary/data-browser/pull/824

**Comment by NoopDog** [comment]
Actually  our updated process is to PR to Databiosphere, then I can sync our fork to get the changes back.

File: explorer/next-sitemap.config.mjs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/** @type {import('next-sitemap').IConfig} */
+const siteMapConfig = {
+  changefreq: "monthly",
+  generateIndexSitemap: false,
+  outDir: "./out/explore",
```
Comment (NoopDog) [inline explorer/next-sitemap.config.mjs]: @BruceRodrigues the outDir depends on the basePath as well and should be process.env.NEXT_PUBLIC_BASE_PATH || "/explore" . Can you update this please? Thanks! D

**Review by NoopDog — CHANGES_REQUESTED** [review]

File: explorer/next-sitemap.config.mjs (hunk 3 above)
Comment (BruceRodrigues) [reply explorer/next-sitemap.config.mjs]: oh right! Sorry about that

**Review by NoopDog — APPROVED** [review]

**Review by NoopDog — APPROVED** [review]
